src/main/scala/io/shiftleft/fuzzyc2cpg/parser/CodeLocationExtractor.scala

```diff
@@ -0,0 +1,18 @@
+package io.shiftleft.fuzzyc2cpg.parser
+
+import io.shiftleft.fuzzyc2cpg.ast.CodeLocation
+import org.antlr.v4.runtime.ParserRuleContext
+
+object CodeLocationExtractor {
+
+  def extractFromContext(ctx: ParserRuleContext): CodeLocation = {
+    val startLine = Some(ctx.start.getLine)
+    val startPos = Some(ctx.start.getCharPositionInLine)
+    val startIndex = Some(ctx.start.getStartIndex)
+    val endLine = if (ctx.stop != null) { Some(ctx.stop.getLine) } else { None }
```

Comment:
```suggestion
    val endLine = Option(ctx.stop.getLine)
```
Same for the others below.